### ISSUE TITLE
Accepting a selection kills the snippet

### ISSUE THREAD

**Issue by ignamartinoli** [Saghen/blink.cmp#373]
### Make sure you have done the following

- [X] I have updated to the latest version of `blink.cmp`
- [X] I have read the README

### Bug Description

This happens if I use LazyVim default config or my own (I would call it a Super-Super-Tab):

```
keymap = {
	['<Down>'] = { 'fallback' },
	['<S-Tab>'] = { 'select_prev', 'snippet_backward', 'fallback' },
	['<Tab>'] = { 'select_next', 'snippet_forward', 'fallback' },
	['<Up>'] = { 'fallback' },
	['<Enter>'] = { 'accept' }
},
windows = {
	autocomplete = {
		selection = 'manual'
	}
}
```

I noticed with `gopls` that when accepting a selection, the snippet engine gets killed and I can no longer move. Also the text suggested by the snippet gets pasted, so I have to manually delete it.

### Relevant configuration

```lua
highlight = {
	use_nvim_cmp_as_default = false
},
nerd_font_variant = "mono",
windows = {
	autocomplete = {
		winblend = vim.o.pumblend
	},
	documentation = {
		auto_show = true
	},
	ghost_text = {
		enabled = vim.g.ai_cmp
	}
},
accept = { auto_brackets = { enabled = true } },
sources = {
	compat = {},
	completion = {
		enabled_providers = { "lsp", "path", "snippets", "buffer" }
	}
},
keymap = {
	preset = "enter",
	["<Tab>"] = {
		LazyVim.cmp.map({ "snippet_forward", "ai_accept" }),
		"fallback",
	}
}
```

### neovim version

NVIM v0.10.2 Build type: Release LuaJIT 2.1.1731601260

### `blink.cmp` version: branch, tag, or commit

v0.6.1

**Comment by Saghen** [comment]
Is the item you selected inserting a snippet? You can double check by adding a `vim.print(resolved_item)` in `~/.local/share/nvim/lazy/blink.cmp/completion/accept/init.lua`, and checking if `insertTextMode = 2` when accepting. If that's the case, there's not much we can do here since neovim doesn't support multiple snippet sessions

**Comment by ignamartinoli** [comment]
I found the file. Where exactly should I place the print?

I recorded this video so that it is easier to understand what happens:

https://github.com/user-attachments/assets/e5a634d8-f41a-421f-90c5-e048b14f1286

As you can see, the `WithTimeout` method begins a snippet (I guess), but if I press `<CR>` to use autocompletion inside it the snippet gets killed and the placeholder text is inserted.

**Comment by Saghen** [comment]
> Where exactly should I place the print?

At the beginning of the function that receives `resolved_item`

**Comment by Saghen** [comment]
Yeah looks like your LSP is marking those completions as snippets, even though they have no tabstops. There's not much we can do on the `blink.cmp` side. The Go LSP should be marking these at `PlainText` instead so you might want to open an issue over there

**Comment by ignamartinoli** [comment]
Thanks for the clear response, guidance and patience. Very nice project. I will open an issue there

**Comment by Parsifa1** [comment]
```lua
        snippets = {
            expand = function(snippet)
                require("luasnip").lsp_expand(snippet)
            end,
            active = function(filter)
                if filter and filter.direction then
                    return require("luasnip").jumpable(filter.direction)
                end
                return require("luasnip").in_snippet()
            end,
            jump = function(direction)
                require("luasnip").jump(direction)
            end,
        },
```
add these, you will get a normal tab jump
